packages/lib-classifier/src/store/SubjectStore/SubjectStore.js

```diff
@@ -176,7 +182,15 @@ const SubjectStore = types
       newSubjects.forEach(subject => {
         try {
           const alreadyStored = self.resources.get(subject.id)
-          if (!alreadyStored) {
+          // assume all subjects from Panoptes are unseen by default
+          let notSeen = true
+          if (self.prioritized) {
+            const metadataPriority = subject.metadata['#priority'] ?? subject.metadata.priority
+            // subject metadata in the API response are strings, not numbers.
+            const priority = metadataPriority ? parseFloat(metadataPriority) : -1
+            notSeen = priority > self.last.priority
+          }
+          if (notSeen && !alreadyStored) {
```

Comment:
The code read here looks good, generally. I have some questions:

- Is there any scenario when `self.last` is undefined? If yes, then `self.last.priority` is going to crash the classifier. However, if self.last should _never_ be undefined, then we should keep this as is if the classifier crashes, 
- If a Subject doesn't have a valid priority (e.g. maybe a project owner mistyped it as `subject.metadata.priority = 'page 1'`), it should never get added to the queue, correct?
  - Note: if the last item in a queue has an invalid priority, it's going to mess up the rest of the queue, as `priority > self.last.priorty` becomes `1234 > NaN` (EDIT: `1234 > undefined` is also possible) which will always evaluate to false
  - Not a problem for this PR per se, but it does mean everyone needs to be aware that it's the project owner's responsibility for setting up their Subjects properly, and there are no safety nets.

Reply:
In either of those cases, `subject.priority` is broken anyway, I guess.
https://github.com/zooniverse/front-end-monorepo/blob/e1c866f28057fdc9bac09f00b8b28eeda0afe6d1/packages/lib-classifier/src/store/SubjectStore/Subject/Subject.js#L93-L99

Reply:
Good question about `subjects.last`. It's undefined when the queue is empty, so when the classifier first loads.
https://github.com/zooniverse/front-end-monorepo/blob/e1c866f28057fdc9bac09f00b8b28eeda0afe6d1/packages/lib-classifier/src/store/SubjectStore/SubjectStore.js#L75-L83

Reply:
I'm fixing that bug right now but here's something else I've found while testing. A workflow can have `workflow.prioritized` set, but use _randomised subject selection_ if that workflow is complete and all the subjects are retired.

Try it out here, with a prioritised, completed workflow from Davy Notebbooks:
https://localhost:8080/?project=humphrydavy/davy-notebooks-project&workflow=19689&env=production&demo=true

Reply:
I've added a check for `user_has_finished_workflow` to cover that case.

Reply:
Noted, let me re-run some tests for that case 👍